overlord/ifacestate/helpers.go

```diff
@@ -540,3 +567,161 @@ func resolveSnapIDToName(st *state.State, snapID string) (name string, err error
 	}
 	return decl.SnapName(), nil
 }
+
+// SnapMapper offers APIs for re-mapping snap names in interfaces and the
+// configuration system. The mapper is designed to apply transformations around
+// the edges of snapd (state interactions and API interactions) to offer one
+// view on the inside of snapd and another view on the outside.
+type SnapMapper interface {
+	// re-map functions for loading and saving objects in the state.
+	RemapSnapFromState(snapName string) string
+	RemapSnapToState(snapName string) string
+	// re-map functions for API requests/responses.
+	RemapSnapFromRequest(snapName string) string
+	RemapSnapToResponse(snapName string) string
+}
+
+// IdentityMapper implements SnapMapper and performs no transformations at all.
+type IdentityMapper struct{}
+
+// RemapSnapFromState doesn't change the snap name in any way.
+func (m *IdentityMapper) RemapSnapFromState(snapName string) string {
+	return snapName
+}
+
+// RemapSnapToState doesn't change the snap name in any way.
+func (m *IdentityMapper) RemapSnapToState(snapName string) string {
+	return snapName
+}
+
+// RemapSnapFromRequest  doesn't change the snap name in any way.
+func (m *IdentityMapper) RemapSnapFromRequest(snapName string) string {
+	return snapName
+}
+
+// RemapSnapToResponse doesn't change the snap name in any way.
+func (m *IdentityMapper) RemapSnapToResponse(snapName string) string {
+	return snapName
+}
+
+// CoreCoreSystemMapper implements SnapMapper and makes implicit slots
+// appear to be on "core" in the state and in memory but as "system" in the API.
+//
+// NOTE: This mapper can be used to prepare, as an intermediate step, for the
+// transition to "snapd" mapper. Using it the state and API layer will look
+// exactly the same as with the "snapd" mapper. This can be used to make any
+// necessary adjustments the test suite.
+type CoreCoreSystemMapper struct {
+	IdentityMapper // Embedding the nil mapper allows us to cut on boilerplate.
```

Comment:
nil mapper seems the wrong name now?  same below

Reply:
Fixed here and below.